meetings/working-groups/discriminated-unions/extended-enums.md

```diff
@@ -0,0 +1,627 @@
+
+It also aims to consolidate the majority of the design space and feedback over many years in this repository, especially from:
+
+- [#113](https://github.com/dotnet/csharplang/issues/113) - Main champion issue for discriminated unions, 7+ years of discussion and multiple LDM meetings
+
+<details>
+  <summary>as well as...</summary>
+
+  - [#75](https://github.com/dotnet/csharplang/issues/75) - Early comprehensive discussion exploring DU syntax options including enum class
+  - [#2962](https://github.com/dotnet/csharplang/discussions/2962) - Major discussion on Andy Gocke's DU proposal, debating enum class vs enum struct
+  - [#7016](https://github.com/dotnet/csharplang/issues/7016) - Fast, efficient unions proposal focusing on struct-based implementations
+  - [#3760](https://github.com/dotnet/csharplang/discussions/3760) - Community "shopping list" of desired discriminated union features
+  - [#7544](https://github.com/dotnet/csharplang/issues/7544) - Simple encoding of unions exploring type unions vs tagged unions
+  - [#8804](https://github.com/dotnet/csharplang/discussions/8804) - String-based enums for cloud services with extensibility needs
+  - [#1860](https://github.com/dotnet/csharplang/issues/1860) - Long-running request for string enum support citing TypeScript/Java
+  - [#9010](https://github.com/dotnet/csharplang/discussions/9010) - "Closed" enum types that guarantee exhaustiveness
+  - [#6927](https://github.com/dotnet/csharplang/discussions/6927) - Constant enums discussion around strict value enforcement
+  - [#7854](https://github.com/dotnet/csharplang/issues/7854) - Exhaustiveness checking for ADT patterns using private constructors
+  - [#8942](https://github.com/dotnet/csharplang/discussions/8942) - Track subtype exhaustiveness for closed hierarchies
+  - [#8926](https://github.com/dotnet/csharplang/discussions/8926) - Extensive discussion on Option<T> as canonical DU use case
+  - [#7010](https://github.com/dotnet/csharplang/discussions/7010) - Union types discussion heavily featuring Option<T> and Result<T>
+  - [#274](https://github.com/dotnet/csharplang/discussions/274) - Java-style class-level enums with methods and constructors
+</details>
+
+## 1. Overview
+
+### Two Complementary Features
+
+C# will gain two separate features for different modeling needs: [type unions](https://raw.githubusercontent.com/dotnet/csharplang/refs/heads/main/proposals/unions.md) and enhanced enums (this proposal). These features work together but solve distinct problems in the type system.
+
+### At a Glance
+
+```csharp
+// Type unions - one value, multiple possible types
+union Result { string, ValidationError, NetworkException }
+
+// Enhanced enums - one type, multiple possible shapes  
+enum PaymentResult
```

Comment:
Aren't enhanced enums a superset of type unions?

Given

```cs
union Result { string, ValidationError, NetworkException }
```

I could just represent it as

```cs
enum struct Result { Text(string value), ValidationError(ValidationError value), NetworkException(NetworkException value) }
```

and the usage of such enhanced enum would be mostly similar to the type union I think. The only difference would be slightly more verbose syntax at the declaration side (but we can always add more syntactic sugar). Hence it'd be enough to have enhanced enums and not type unions?

Reply:
Yes. There is an argument to be made here that unions are just a very very terse way of annotating a value to say it is one of a few types. And yes, it would be a good question to ask if that is valuable enough. 

Personally,I think it is. But the conversation is a good one:-)